tools/converter/osm2cityflow.py

```diff
@@ -0,0 +1,508 @@
+import argparse
+import json
+import xml.dom.minidom
+import copy
+import pyproj as proj
+import folium
+
+intersections = []
+roads = []
+node_dic = {}
+way_dic = {}
+road_dic = {}
+lanes_dic = {}
+way_direction = {}
+cross = {}
+min_lat = 1000.0
+min_lon = 1000.0
+
+
+def geo_converter(lon, lat):
+    crs_wgs = proj.Proj(init='epsg:4326')
+    crs_bng = proj.Proj(init='epsg:8826')
+
+    x1, y1 = proj.transform(crs_wgs, crs_bng, lon, lat)
+    x2, y2 = proj.transform(crs_wgs, crs_bng, min_lon, min_lat)
+    return (x1 - x2), (y1 - y2)
+
+
+def build_virtual_intersection(nodeID, id):
+    node = node_dic[nodeID]
+    x, y = geo_converter(float(node.getAttribute('lon')), float(node.getAttribute('lat')))
+    intersection = {
+        "id": nodeID,
+        "point": {"x": x, "y": y},
+        "width": 0,
+        "roads": [id],
+        "roadLinks": [],
+        "trafficLight": {
+            "roadLinkIndices": [],
+            "lightphases": []
+        },
+        "virtual": True
+    }
+    intersections.append(intersection)
+
+
+def build_road(wayID, maxSpeed, lanes, points, id, startId, endID):
+    road = {
+        "id": id,
+        "points": points,
+        "lanes": [
+        ],
+        "startIntersection": startId,
+        "endIntersection": endID
+    }
+    lane_template = {
+        "width": 3.7,
+        "maxSpeed": maxSpeed
+    }
+    for _ in range(lanes):
+        road["lanes"].append(lane_template)
+    roads.append(road)
+    road_dic[id] = road
+    lanes_dic[id] = lanes
+    way = way_dic[wayID]
+    ndlist = way.getElementsByTagName('nd')
+    if endID == ndlist[0].getAttribute('ref') or endID == ndlist[len(ndlist) - 1].getAttribute('ref'):
+        if endID not in cross or len(cross[endID]) <= 1:
+            build_virtual_intersection(endID, id)
+    if startId == ndlist[0].getAttribute('ref') or startId == ndlist[len(ndlist) - 1].getAttribute('ref'):
+        if startId not in cross or len(cross[startId]) <= 1:
+            build_virtual_intersection(startId, id)
+
+
+def get_tag(taglist):
+    numLanes = -1
+    oneDirectionLanes = -1
+    otherDirectionLanes = -1
+    oneway = False
+    maxSpeed = 0
+    for tag in taglist:
+        if tag.getAttribute('k') == 'lanes':
+            numLanes = int(tag.getAttribute('v'))
+        if tag.getAttribute('k') == 'lanes:forward':
+            oneDirectionLanes = int(tag.getAttribute('v'))
+        if tag.getAttribute('k') == 'lanes:backward':
+            otherDirectionLanes = int(tag.getAttribute('v'))
+        if tag.getAttribute('k') == 'oneway' and tag.getAttribute('v') == 'yes':
+            oneway = True
+        if tag.getAttribute('k') == 'maxspeed':
+            maxSpeed = float(tag.getAttribute('v')[0:-3]) * 1.609344
```

Comment:
why `getAttribute('k')` so many times?

Reply:
there is a list of tags in one road, their order is not fixed.